### PR TITLE
Add Buildkite

### PR DESCRIPTION
**Pull request by jkmassel**
Adds Buildkite to the project, and removes CircleCI. Also:
- Adds Rubocop and fixes related issues.
- Adds Fastlane to run tests.
- Replaces use of Carthage with SwiftPM.
- Disables two tests that didn't work (/cc @leandroalonso)

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  # Common environment values to use with the `env` key.
+  env: &common_env
+    IMAGE_ID: xcode-13
+
+# This is the default pipeline – it will build and test the app
+steps:
+  #################
+  # Build and Test
+  #################
+  - label: "🧪 Build and Test"
```
Comment (mokagio) [inline .buildkite/pipeline.yml]: Nitpick: 
```suggestion
  - label: "🔬 Build and Test"
```

File: Gemfile.lock [hunk 2]
```diff
@@ -7,10 +7,31 @@ GEM
...
+      aws-sigv4 (~> 1.1)
+    aws-sdk-s3 (1.112.0)
+      aws-sdk-core (~> 3, >= 3.126.0)
+      aws-sdk-kms (~> 1)
+      aws-sigv4 (~> 1.4)
+    aws-sigv4 (1.4.0)
+      aws-eventstream (~> 1, >= 1.0.2)
+    babosa (1.0.4)
     claide (1.0.3)
     cocoapods (1.8.4)
```
Comment (mokagio) [inline Gemfile.lock:36]: The latest CocoaPods version at the time of this comment is 1.11.2. Should we take the chance of this PR to upgrade?

The same likely goes for Fastlane, too.

File: MediaEditor.podspec [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
+
 Pod::Spec.new do |s|
   s.name          = 'MediaEditor'
   s.version       = '1.2.1'
 
   s.summary       = 'An extensible Media Editor for iOS.'
   s.description   = <<-DESC
                     An extensible Media Editor for iOS that allows editing single or multiple images.
-                  DESC
+  DESC
```
Comment (mokagio) [inline MediaEditor.podspec]: CocoaPods likely does some post processing on this, but the heredoc with the `-` will keep all the spaces used for indentation here:

image

We could either remove the spaces or use the `<<~` heredoc, which strips the left spaces.

In the latter, Rubocop tells me to use two spaces for indentation, so:

```ruby
Pod::Spec.new do |s|
  # ...
  s.description   = <<~DESC
    An extensible Media Editor for iOS that allows editing single or multiple images.
  DESC
```

File: fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+default_platform(:ios)
+
+platform :ios do
+  desc 'Builds the project and runs tests'
+  lane :test do
+    run_tests(
+      project: 'MediaEditor.xcodeproj',
+      scheme: 'MediaEditor',
+      devices: ['iPhone 11'],
+      deployment_target_version: '14.5',
```
Comment (mokagio) [inline fastlane/Fastfile:12]: Have you tried the latest for Xcode 13.0 (used in CI), which I guess would be 15?

Comment (mokagio) [reply fastlane/Fastfile:12]: FWIW, I tried this on my machine and it worked

```suggestion
      deployment_target_version: '15.0',
```

File: MediaEditor.xcodeproj/xcshareddata/xcschemes/MediaEditor.xcscheme [hunk 5]
```diff
@@ -28,6 +28,24 @@
...
+               ReferencedContainer = "container:MediaEditor.xcodeproj">
+            </BuildableReference>
+            <SkippedTests>
+               <Test
+                  Identifier = "MediaEditorHubTests/testDoNotShowActivityIndicatorIfImageIsNotBeingLoaded()">
+               </Test>
+               <Test
+                  Identifier = "MediaEditorHubTests/testShowActivityIndicatorWhenSwipingToAnImageBeingLoaded()">
+               </Test>
+            </SkippedTests>
```
Comment (mokagio) [inline MediaEditor.xcodeproj/xcshareddata/xcschemes/MediaEditor.xcscheme:47]: Whilst skipping these is a temporary tech debt incurrence, I'd like to suggest we use `XCTExpectFailure` instead.

I opened #29 for this and added a rationale.

---

TIL that the noun deriving from the verb _incur_ is _incurrence_, not _incursion_. However, "tech debt incursion" might still be appropriate ⚔️ ⚔️ ⚔️

**Review by mokagio — APPROVED** [review]
I checked out this branch locally and verified the SPM setup works (which to be fair was sort of unnecessary, given that CI built fine)

Happy to see you already marked the CI tasks as required 👍 

image

My suggestion in #29 is not required for this to be merged into `trunk`.

File: MediaEditor.xcodeproj/xcshareddata/xcschemes/MediaEditor.xcscheme (hunk 5 above)
Comment (jkmassel) [reply MediaEditor.xcodeproj/xcshareddata/xcschemes/MediaEditor.xcscheme:47]: > tech debt incursion
😱 😂

File: fastlane/Fastfile (hunk 4 above)
Comment (jkmassel) [reply fastlane/Fastfile:12]: So...as identified in https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1349/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688R37, this value really doesn't do much (it'll use iOS 15 under the hood no matter what this is set to).

I figure we can run around and update all our projects once:
1. That PR lands
2. I'm able to extract the simulator creation bits into the release toolkit, then update all our projects to validate the available simulator versions.

File: MediaEditor.podspec (hunk 3 above)
Comment (jkmassel) [reply MediaEditor.podspec]: Ah nice one! Addressed in ce96d6fe0e5282841184383d842f89a5b4eb501b

**Comment by jkmassel** [comment]
Merging this to unblock #29, and I'll post a followup soon to address the simulator stuff Gio mentioned.

**Pull request merged**